resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -2,7 +2,7 @@
 jruby-puppet: {
     # Where the puppet-agent dependency places puppet, facter, etc...
     # Puppet server expects to load Puppet from this location
-    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby, /opt/puppetlabs/puppet/cache/lib]
```

Comment:
@joshcooper Back to the discussion, I'm a bit confused at how switching to an absolute path in require statements would solve this issue.  What is the value of `__FILE__` in your example?

Are you thinking we'd return to a world where we load the code directly from the module directory structure, or are you thinking that the absolute path wouldn't be relative to `__FILE__` but rather relative to the agent's plugin sync directory?

Maybe something like hiera doing a `require "#{Puppet[:libdir]}/hiera/backends/#{backend_name}.rb"`?